File: workflow/plugins/python/string/string_lower/string_lower.py
"""Workflow plugin: convert string to lowercase."""

from ...base import NodeExecutor


class StringLower(NodeExecutor):
    """Convert string to lowercase."""

    node_type = "string.lower"
    category = "string"
    description = "Convert string to lowercase"

    def execute(self, inputs, runtime=None):
        value = str(inputs.get("value", inputs.get("text", "")))
        return {"result": value.lower()}

File: workflow/plugins/go/string/string_lower/string_lower.go
// Package string_lower provides a workflow plugin for lowercasing strings.
package string_lower

import (
	"strings"
)

// StringLower implements the NodeExecutor interface for lowercasing strings.
type StringLower struct {
	NodeType    string
	Category    string
	Description string
}

// NewStringLower creates a new StringLower instance.
func NewStringLower() *StringLower {
	return &StringLower{
		NodeType:    "string.lower",
		Category:    "string",
		Description: "Convert string to lowercase",
	}
}

// Execute runs the plugin logic.
func (p *StringLower) Execute(inputs map[string]interface{}, runtime interface{}) map[string]interface{} {
	str, ok := inputs["string"].(string)
	if !ok {
		return map[string]interface{}{"result": "", "error": "string is required"}
	}

	return map[string]interface{}{"result": strings.ToLower(str)}
}
